### PR TITLE
Fix inconsistency when build both shared/static lib

### PR DESCRIPTION
**Pull request by dota17**
Hi @SpaceIm please check this patch if it solves your issue.

### CODE REVIEW

File: src/lib_json/CMakeLists.txt [hunk 1]
```diff
@@ -151,7 +151,11 @@ if(BUILD_SHARED_LIBS)
...
 if(BUILD_STATIC_LIBS)
-    set(STATIC_LIB ${PROJECT_NAME}_static)
+    if (BUILD_SHARED_LIBS)
+        set(STATIC_LIB ${PROJECT_NAME}_static)
+    else()
+        set(STATIC_LIB ${PROJECT_NAME}_lib)
+    endif()
     add_library(${STATIC_LIB} STATIC ${PUBLIC_HEADERS} ${JSONCPP_SOURCES})
 
     # avoid name clashes on windows as the shared import lib is alse named jsoncpp.lib
```
Comment (xkszltl) [inline src/lib_json/CMakeLists.txt:161]: typo: alse -> also

File: src/lib_json/CMakeLists.txt [hunk 2]
```diff
@@ -160,7 +164,7 @@ if(BUILD_STATIC_LIBS)
     endif()
 
     set_target_properties(${STATIC_LIB} PROPERTIES
-        OUTPUT_NAME jsoncpp${STATIC_SUFFIX}
+        OUTPUT_NAME jsoncpp
```
Comment (xkszltl) [inline src/lib_json/CMakeLists.txt:167]: I would recommend to keep them different (and consistent across versions) to support packaging of dual (shared+static) build.
As you may noticed MSVC generates import lib (`.lib`) for `.dll`, which should not be confused with static `.lib`.

File: src/lib_json/CMakeLists.txt [hunk 3]
```diff
@@ -151,7 +151,11 @@ if(BUILD_SHARED_LIBS)
 endif()
 
 if(BUILD_STATIC_LIBS)
-    set(STATIC_LIB ${PROJECT_NAME}_static)
+    if (BUILD_SHARED_LIBS)
+        set(STATIC_LIB ${PROJECT_NAME}_static)
```
Comment (xkszltl) [inline src/lib_json/CMakeLists.txt:155]: You can `add_library(<default name> ALIAS jsoncpp_static)` or `add_library(<default name> INTERFACE)` and them `target_link_libraries(<default name> INTERFACE jsoncpp_static)

**Review by xkszltl — CHANGES_REQUESTED** [review]

**Comment by baylesj** [comment]
Closing this due to inactivity. Feel free to reopen if you are able to address feedback.

**Pull request closed without merging**